src/supervised/other.jl

```diff
@@ -1,21 +1,101 @@
 @doc doc"""
-    MisclassLoss <: SupervisedLoss
+    MisclassLoss{R<:AbstractFloat} <: SupervisedLoss
 
 Misclassification loss that assigns `1` for misclassified
 examples and `0` otherwise. It is a generalization of
 `ZeroOneLoss` for more than two classes.
-"""
-struct MisclassLoss <: SupervisedLoss end
-
-agreement(target, output) = target == output
-
-value(::MisclassLoss, agreement::Bool) = agreement ? 0 : 1
-deriv(::MisclassLoss, agreement::Bool) = 0
-deriv2(::MisclassLoss, agreement::Bool) = 0
 
-value(loss::MisclassLoss, target::Number, output::Number) = value(loss, agreement(target, output))
-deriv(loss::MisclassLoss, target::Number, output::Number) = deriv(loss, agreement(target, output))
-deriv2(loss::MisclassLoss, target::Number, output::Number) = deriv2(loss, agreement(target, output))
+The type parameter `R` specifies the result type of the
+loss. Default type is double precision `R = Float64`.
+"""
+struct MisclassLoss{R<:AbstractFloat} <: SupervisedLoss end
```

Comment:
I like the parameterization 👍

Reply:
Thanks :) In the next round of improvements, I plan to write more tests covering the result type of `value`, `deriv`, and `deriv2`.